### PR TITLE
Make OkResponse non-generic

### PR DESCRIPTION
**Pull request by nguerrera**
Fix https://github.com/Azure/cadl-azure/issues/545

### CODE REVIEW

File: packages/rest/lib/http.cadl [hunk 1]
```diff
@@ -9,13 +9,18 @@ model Response<Status> {
...
-model OkResponse<Body> is Response<200> {
-  @doc("The reponse body.")
+
+/**
+ * Defines a model with a single property of the given type, marked with `@body`.
+ *
+ * This can be useful in situations where you cannot use a bare T as the body
+ * and it is awkward to add a property.
+ */
+model Body<T> {
```
Comment (daviwil) [inline packages/rest/lib/http.cadl:19]: Would `Payload<T>` be more obvious or even less obvious?

**Review by daviwil — APPROVED** [review]
Looks great!  The metadata changes are really helping use Cadl's syntax to the fullest.  The intersection operator is way more useful and expressive now.

**Comment by nguerrera** [comment]
I'll write up the breaking change notes for review before merging this.

File: packages/rest/lib/http.cadl (hunk 1 above)
Comment (nguerrera) [reply packages/rest/lib/http.cadl:19]: I think since this is in HTTP library, there's nothing to gain for adding a term other than Body.

**Comment by nguerrera** [comment]
Wrote this: https://github.com/microsoft/cadl/issues/728#issuecomment-1197327245

**Review by timotheeguerin — APPROVED** [review]

**Pull request merged**